core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -170,27 +170,37 @@ contract Liquidatable is PricelessPositionManager {
      * for the sponsor.
-     * @param sponsor address to liquidate
-     * @param amountToLiquidate amount of liquidatedCollateral that liquidator wants to liquidate
      * @return uuid of the newly created liquidation
      */
-    function createLiquidation(address sponsor, FixedPoint.Unsigned calldata amountToLiquidate)
-        external
-        fees()
-        onlyPreExpiration()
-        returns (uint uuid)
-    {
-        // Attempt to retrieve Position data for sponsor
+    // TODO: Perhaps pass this ID via an event rather than a return value
+    function createLiquidation(
+        address sponsor,
+        FixedPoint.Unsigned calldata collateralPerToken,
+        FixedPoint.Unsigned calldata maxTokensToLiquidate
+    ) external fees() onlyPreExpiration() returns (uint uuid) {
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
-        FixedPoint.Unsigned memory positionCollateral = _getCollateral(positionToLiquidate.rawCollateral);
-        require(positionCollateral.isGreaterThan(0));
 
-        // Caller is required to include in order to prevent front-running attacks that modify the liquidatedCollateral amount
-        // @dev amountToLiquidate is >= underlying collateral because the position is still liquidatable if the underlying collateral declines
-        // while the liquidation transaction is pending
-        require(
-            amountToLiquidate.isGreaterThanOrEqual(positionCollateral.sub(positionToLiquidate.withdrawalRequestAmount))
+        // tokensToLiquidate = minimum of maxTokensToLiquidate and how many tokens are outstanding.
+        FixedPoint.Unsigned memory tokensToLiquidate = FixedPoint.Unsigned(maxTokensToLiquidate.rawValue).isGreaterThan(
+            positionToLiquidate.tokensOutstanding
+        )
+            ? positionToLiquidate.tokensOutstanding
+            : FixedPoint.Unsigned(maxTokensToLiquidate.rawValue);
+        // What ratio of the Position is getting liquidated, will be used to scale everything else.
+        FixedPoint.Unsigned memory ratio = tokensToLiquidate.div(positionToLiquidate.tokensOutstanding);
+
+        // DO SOME CHECKS HERE IN A SEPARATE FUNCTION.
+        FixedPoint.Unsigned memory startCollateral = _getCollateral(positionToLiquidate.rawCollateral);
+        FixedPoint.Unsigned memory startCollateralNetOfWithdrawal = startCollateral.sub(
+            positionToLiquidate.withdrawalRequestAmount
         );
+        FixedPoint.Unsigned memory startTokens = positionToLiquidate.tokensOutstanding;
+
+        // The actual amount of collateral that gets moved to the liquidation.
+        FixedPoint.Unsigned memory lockedCollateral = startCollateral.mul(ratio);
+        // For purposes of disputes, it's actually this liquidatedCollateral value that's used. This value is net of
+        // withdrawal requests.
+        FixedPoint.Unsigned memory liquidatedCollateral = startCollateralNetOfWithdrawal.mul(ratio);
```

Comment:
I'm not 100% confident this actually does anything, but I think `liquidatedCollateral` should be `lockedCollateral.sub(positionToLiquidate.withdrawalRequestAmount` so that there is one less `mul` operation which has precision loss possibility

Reply:
I *think* those are different because presumably the withdrawalRequestAmount also needs to get scaled down by ratio. Not 100% sure though.

Reply:
Ok, but where do you scale down `positionToLiquidate.withdrawalRequestAmount`?

Reply:
I think that makes sense. I think it might even be cleaner to perform the scaling down of the `withdrawalRequestAmount` (currently done on line `474`) here (i.e. `FixedPoint.Unsigned memory withdrawalAmountToRemove = positionData.withdrawalRequestAmount.mul(ratio)`), and then pass in `withdrawalAmountToRemove` into `reduceSponsorPosition`. 

This way you don't need to pass in `ratio` and groups all of the scaling down logic together and makes all of the reducePosition logic the same

Reply:
Makes sense, will refactor accordingly.